quattor/functions/hardware.pan

```diff
@@ -1,39 +1,92 @@
-
 declaration template quattor/functions/hardware;
 
 
-# Returns the number of cores on the current machine or another machine
-# specifified as a HW template (string) or a HW configuration (nlist)�.
-function get_num_of_cores = {
+@documentation{
+    descr = Returns the hardware tree of the current machine or another machine
+    arg = A HW template (string) or a HW configuration (nested dict) to use as a source of hardware configuration (optional)
+}
+function get_hw_config = {
     if ( ARGC == 0 ) {
-      hw_config = value('/hardware');
+        hw_config = value('/hardware');
     } else if ( ARGC == 1 ) {
-      if ( is_nlist(ARGV[0]) ) {
-        hw_config = ARGV[0];
-      } else if ( is_string(ARGV[0]) ) {
-        hw_config = create(ARGV[0]);
-      } else {
-        error(format('Invalid argument type (%s)',to_string(ARGV[0])));
-      }
+        if ( is_dict(ARGV[0]) ) {
+            hw_config = ARGV[0];
+        } else if (is_string(ARGV[0])) {
+            hw_config = create(ARGV[0]);
+        } else {
+            error('Invalid argument type (%s)', to_string(ARGV[0]));
+        }
     } else {
-      error(format('get_num_of_cores requires 0 or 1 argument (%s specified)',ARGC));
+        error('get_num_of_cores requires 0 or 1 argument (%s specified)', ARGC);
```

Comment:
Use FUNCTION here to avoid the typo in the error message?  (This is no longer get_num_of_cores)